fix for extra spaces in the beginning every page of handbook.

tirukovelamanoj opened the pull request:
**Description**

This PR fixes #
[Handbook] Extra spaces in the beginning every page of handbook. #6312

**Notes for Reviewers**

Total 16 pages in Table of Contents:

1. [About](https://layer5.io/community/handbook/contributor-journey)
2. [Guidelines](https://layer5.io/community/handbook/community)
3. [Roles](https://layer5.io/community/handbook/community-roles)
4. [Contribution](https://layer5.io/community/handbook/contribution)
5. [Recognition](https://layer5.io/community/handbook/recognition)
6. [Repository Overview](https://layer5.io/community/handbook/repository-overview)
7. [Projects](https://layer5.io/community/handbook/projects)
8. [Mentorship Programs](https://layer5.io/community/handbook/mentorship-programs)
9. [Writing Program](https://layer5.io/community/handbook/writing-program)
10. [UX Contributors](https://layer5.io/community/handbook/designer)
11. [Learning](https://layer5.io/community/handbook/learn-layer5)
12. [Connect with us](https://layer5.io/community/handbook/connect-with-us)
13. [Code of Conduct](https://layer5.io/community/handbook/code-of-conduct)
14. [Security Vulnerabilities](https://layer5.io/community/handbook/security-vulnerabilities)
15. [GitHub Process](https://layer5.io/community/handbook/github-process)
16. [FAQs](https://layer5.io/community/handbook/faq)

And there is extra notable space when the screen width is `750px-768px`  and  `>768px` for the below pages (Group 1):

1. [Contribution](https://layer5.io/community/handbook/contribution)
2. [Projects](https://layer5.io/community/handbook/projects)
3. [Writing Program](https://layer5.io/community/handbook/writing-program)
4. [UX Contributors](https://layer5.io/community/handbook/designer)
5. [Code of Conduct](https://layer5.io/community/handbook/code-of-conduct)
6. [Security Vulnerabilities](https://layer5.io/community/handbook/security-vulnerabilities)
7. [GitHub Process](https://layer5.io/community/handbook/github-process)
8. [FAQs](https://layer5.io/community/handbook/faq)

These can be fixed by adjusting the `margin-top` css property of `page-section` class in `Handbook.style.js`

```
      @media (min-width: 750px) {
        margin-top: -32rem;
        margin-left: 20rem;
      }
```

```
      @media screen and (min-width: 768px) and (max-height: 1145px) {
        margin-top : -62vh ;
      }
```

And below are the pages that are having notable extra space when the screen width is in between `750-768px` (Group 2):

1. [About](https://layer5.io/community/handbook/contributor-journey)
2. [Guidelines](https://layer5.io/community/handbook/community)
3. [Roles](https://layer5.io/community/handbook/community-roles)
4. [Repository Overview](https://layer5.io/community/handbook/repository-overview)
5. [Learning](https://layer5.io/community/handbook/learn-layer5)
6. [Connect with us](https://layer5.io/community/handbook/connect-with-us)

These can also be fixed by adjusting the same CSS properties as above but the reason we are not seeing the extra notable space in-between `750-768px` screen width is due to the following lines in code

```
          <Container>
            <div className="content">
              <a id="top">
                <h2>Layer5 community leader undertake different roles</h2>
              </a>
              <p>
                A leader is someone who can contribute to the Layer5 Community's
                growth by faithfully upholding one of the responsibilities outlined in the roles below.
              </p>
```

![Image](https://github.com/user-attachments/assets/86ec40fd-8e3f-4a56-9d63-d30090afff8a)

So because of `<a><h2></h2><p></p><a>` these tags being inside the `<div className="content">`, the problem is becomming different if we move them above the div tag like the below these would also fall under the same fix as of first 8 pages (Group 1)

```
        <Container>
          <h2 className="heading-top">General contribution flow</h2>
          <p>
            Pull requests (PRs) are the best ways to propose changes to a
            project repository. At Layer5 org, we use the Github Flow:
          </p>
          <div className="content">
            <a id="Clone your fork">  <h3>Clone your fork to your local machine</h3> </a>
            <ul>
```

![Image](https://github.com/user-attachments/assets/27b0fe60-9fd9-4bf4-9779-4e34e11e24da)

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

M-DEV-1 commented:
Hi @tirukovelamanoj 👋🏼 

The reduction is extra spaces is great, but we'll need to reduce them a bit more, if possible. Some of these pages have other broken links or images, but we can deal with them in a seperate issue.

![image](https://github.com/user-attachments/assets/ee779117-eebd-41c6-bdbb-effbda38a23c)

M-DEV-1 commented:
@tirukovelamanoj can we make these changes by manipulating the `content` class? instead of manually pushing these elements outside the div/

LibenHailu commented:
![image](https://github.com/user-attachments/assets/94852d10-b39c-49d8-b392-c833312b39ee)
![Screenshot from 2025-06-04 10-22-43](https://github.com/user-attachments/assets/2ecd690f-de31-4597-b1d6-4b6dd105ea90)

l5io commented:
🚀 Preview for commit 0e91acd7326fc5a382ed9ec29684e50a1040f425 at: https://68445dbb9928c8331183811f--layer5.netlify.app

l5io commented:
🚀 Preview for commit 9662f371d347d5a91b7ca0d45bf65ae01035e224 at: https://6844ef30650af7c13fe3c509--layer5.netlify.app

tirukovelamanoj commented:
Thank you @M-DEV-1 , @LibenHailu for the review.

I attempted to use the selective rendering method mentioned in the post above. While it did reduce the overall build time, the responsiveness was still lacking, and page rendering remained slow. It took nearly 15 minutes just to load the home page. I tested this approach both on my local machine and in GitHub Codespaces, but was only able to successfully complete the build once while using Codespaces.

@M-DEV-1 as requested, I have updated the CSS to properly display the content section along with the heading and text. I also made some code adjustments on a few other pages, moving the `<h2>` and `<p>` tags inside the content `<div>` to apply the same styling changes consistently. Additionally, I fixed issues with broken text and spacing. All 16 pages are now rendering without any noticeably excessive spacing. Please let me know if these changes sufficiently address the issue, and I hope you can review them again.

Please see below for the most recent preview of the changes:
 - https://6844ef30650af7c13fe3c509--layer5.netlify.app/

LibenHailu commented:
Hi @tirukovelamanoj,
Thank you for working under these circumstances. We hope to find a solution soon.

As for the review  it looks good visually. However, it would be preferable to fix the behavior that’s pushing the content down, rather than pulling the content up. It's probably due to a margin-top on some of the elements. I believe this could become an issue in the next iteration. LGTM!

LibenHailu commented:
@tirukovelamanoj Let's create issues for further enhancement.

vr-varad commented:
@tirukovelamanoj 
Thank you for your contribution!
Let's discuss this during the website call today at 7 AM CT

Add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk), if you would :)

leecalcote commented:
@tirukovelamanoj, did you present this work? What was the outcome? Are you still working on this PR?

tirukovelamanoj commented:
> @tirukovelamanoj, did you present this work? What was the outcome? Are you still working on this PR?

@leecalcote, I Apologize for the delay in progressing with the PR. I have been caught up with a lot of personal work lately and, unfortunately, couldn't make time to complete the fixes as planned.

I am still working on the PR and will have it ready for review by tomorrow's meeting. I will incorporate any recommendations that come up and aim to complete it right after. Thanks for your patience, and I appreciate your understanding.

l5io commented:
🚀 Preview for commit 95c101914fe87179101cfefeefba475103cc7636 at: https://686cad530e652570be8b8be6--layer5.netlify.app

l5io commented:
🚀 Preview for commit 387c2e667067b123c7ed184e6ee2fc8793439154 at: https://686cbcdcfab5ec72de9c0a47--layer5.netlify.app
